### PR TITLE
feat: Adds possibility of running DU in the simulation mode

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

This PR allows for running the DU in the simulation mode.
Simulation mode should be used for automated tests and can be enabled by setting the `simulation-mode` charm config option to `True`.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have bumped the version of the library

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -67,6 +67,10 @@ config:
       type: int
       default: 1
       description: Tracking Area Code
+    simulation-mode:
+      type: boolean
+      default: false
+      description: Run DU in simulation mode
```
Comment (gruyaume) [inline charmcraft.yaml:73]: Can you add more info as to what this means?

File: src/templates/du.conf.j2 [hunk 2]
```diff
@@ -158,4 +150,13 @@ log_config: {
     mac_log_level      = "info";
     rlc_log_level      = "info";
     f1ap_log_level     = "info";
-};
\ No newline at end of file
+};
+{% if simulation_mode %}
+rfsimulator: {
+    serveraddr = "server";
```
Comment (gruyaume) [inline src/templates/du.conf.j2:156]: Is this an address? Will this work?

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: src/templates/du.conf.j2 (hunk 2 above)
Comment (Gmerold) [reply src/templates/du.conf.j2:156]: Not sure what that is exactly, but I've tested the workload with this config file and it worked.

File: charmcraft.yaml (hunk 1 above)
Comment (Gmerold) [reply charmcraft.yaml]: Description updated.

**Review by gruyaume — APPROVED** [review]

**Review by gatici — APPROVED** [review]

**Pull request merged**